src/antsibull_docs/schemas/docs/plugin.py

```diff
@@ -208,16 +206,17 @@ class PluginExamplesSchema(BaseModel):
     examples: str = ""
     examples_format: str = "yaml"
 
-    @p.validator("examples_format", always=True)
-    # pylint:disable=no-self-argument
-    def extract_examples_format(cls, value: t.Any, values: dict[str, t.Any]):
-        if isinstance(examples := values.get("examples"), str):
-            if fmt_match := _EXAMPLES_FMT_RE.match(examples.lstrip()):
-                value = fmt_match.group(1)
-        return value
+    @p.model_validator(mode="before")
```

Comment:
Why not use a `field_validator`? That the equivalent of the old `validator` API.

Reply:
I wasn't able to get hold of the other field's values in a `field_validator`.

Also conceptually field validator doesn't really make sense since this sets another field based on a field's value, which is not what a field validator should be doing.

Reply:
:+1: